components/org.wso2.carbon.identity.application.authz.xacml/src/test/java/org.wso2.carbon.identity.application.authz.xacml/pip/AuthenticationContextAttributePIPTest.java

```diff
@@ -0,0 +1,265 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authz.xacml.pip;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockObjectFactory;
+import org.powermock.reflect.internal.WhiteboxImpl;
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.balana.attr.AttributeValue;
+import org.wso2.balana.attr.BagAttribute;
+import org.wso2.balana.cond.EvaluationResult;
+import org.wso2.balana.ctx.EvaluationCtx;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticationRequest;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.authz.xacml.constants.XACMLAppAuthzConstants;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.spy;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * AuthenticationContextAttributePIPTest defines unit tests for AuthenticationContextAttributePIP class.
+ */
+@PrepareForTest({FrameworkUtils.class, LogFactory.class})
+public class AuthenticationContextAttributePIPTest {
+
+    private AuthenticationContextAttributePIP authenticationContextAttributePIP;
+    private EvaluationCtx evaluationCtx;
+    private URI attributeType;
+    private URI attributeId;
+    private URI category;
+    private String issuer;
+    private Log log = mock(Log.class);
+    private static final String TYPE = "type";
+    private static final String CONTEXT_ID = "context_id";
+    private static final String URI_STRING_1 = "12345";
+    private static final String URI_STRING_2 = "12345/12345";
+    private static final String URI_STRING_3 = "//:12345";
+    private static final String ISSUER = "travelocity.com";
+    private static final String USER_IP = "user-ip";
+    private static final String ATTRIBUTE_VALUE = "attributeValue";
+    private static final String HEADER = "header";
+    private static final String AUTHENTICATION_CONTEXT_ATTRIBUTE_PIP = "AuthenticationContextAttributePIP";
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
```

Comment:
This is fine. If you want to get rid of this you can extend PowerMockTestCase :wink: